test/e2e/proxy_test.go

```diff
@@ -72,4 +72,57 @@ var _ = ginkgo.Describe("Proxy [LinuxOnly] [AKSSoakOnly] [Exclude:Arc]", func()
 			}, framework.PollShortTimeout, framework.Poll).Should(gomega.BeTrue())
 		}
 	})
+
+	// This test is to validate the proxy sidecar fallback behavior to AZURE_CLIENT_ID when the client_id parameter is not part of the request.
+	ginkgo.It("should get a valid AAD token after injecting proxy init container and sidecar with no client_id in request", func() {
+		clientID, ok := os.LookupEnv("APPLICATION_CLIENT_ID")
+		gomega.Expect(ok).To(gomega.BeTrue(), "APPLICATION_CLIENT_ID must be set")
+		// trust is only set up for 'proxy-test-sa' service account in the default namespace for now
+		const namespace = "default"
+		serviceAccount := createServiceAccount(f.ClientSet, namespace, "proxy-test-sa", map[string]string{webhook.UseWorkloadIdentityLabel: "true"}, map[string]string{webhook.ClientIDAnnotation: clientID})
+		defer f.ClientSet.CoreV1().ServiceAccounts(namespace).Delete(context.TODO(), serviceAccount, metav1.DeleteOptions{})
+
+		proxyAnnotations := map[string]string{
+			webhook.InjectProxySidecarAnnotation: "true",
+			webhook.ProxySidecarPortAnnotation:   "8080",
+		}
+
+		pod := generatePodWithServiceAccount(
+			f.ClientSet,
+			namespace,
+			serviceAccount,
+			"mcr.microsoft.com/azure-cli",
+			nil,
+			// no client_id in request
+			[]string{"/bin/sh", "-c", "az login -i --allow-no-subscriptions --debug; sleep 3600"},
+			nil,
+			proxyAnnotations,
+			true,
+		)
+
+		pod, err := createPod(f.ClientSet, pod)
+		framework.ExpectNoError(err, "failed to create pod %s in %s", pod.Name, namespace)
+		defer f.ClientSet.CoreV1().Pods(namespace).Delete(context.TODO(), pod.Name, metav1.DeleteOptions{})
+
+		// output proxy and proxy init logs for debugging
```

Comment:
Should this only happen when the test has failed?

Reply:
We have been dumping logs for success/failure runs so the logs are available for reference. The webhook logs are uploaded to artifacts at the end of the test run.